File: VB/ColumnSpacing/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid


Namespace ColumnSpacing
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
			myGridControl1.DataSource = CreateTable(20)
			myGridView1.Columns(1).ColumnSpacing = 8
			myGridView1.BestFitColumns()
		End Sub

		Private Function CreateTable(ByVal RowCount As Integer) As DataTable
			Dim tbl As New DataTable()
			tbl.Columns.Add("Name", GetType(String))
			tbl.Columns.Add("ID", GetType(Integer))
			tbl.Columns.Add("Number", GetType(Integer))
			tbl.Columns.Add("Date", GetType(DateTime))
			For i As Integer = 0 To RowCount - 1
				tbl.Rows.Add(New Object() { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) })
			Next i
			Return tbl
		End Function

		Private Sub myGridView1_RowCellStyle(ByVal sender As Object, ByVal e As RowCellStyleEventArgs) Handles myGridView1.RowCellStyle
			If cbRowCellStyle.Checked Then
				Dim view = myGridView1
				If e.RowHandle <> view.FocusedRowHandle Then
					If e.RowHandle Mod 2 = 0 Then
						e.Appearance.BackColor = Color.FromArgb(&HEB, &HF9, &HEB)
					Else
						e.Appearance.BackColor = Color.FromArgb(&HED, &HF9, &HFA)
					End If
				Else
					e.Appearance.BackColor = SystemColors.GradientInactiveCaption
				End If
			End If
		End Sub

		Private Sub myGridView1_FocusedColumnChanged(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.FocusedColumnChangedEventArgs) Handles myGridView1.FocusedColumnChanged
			btnEdit_colSpacing.EditValue = (TryCast(e.FocusedColumn, MyGridColumn)).ColumnSpacing
		End Sub

		Private Sub btnEdit_colSpacing_ButtonClick(ByVal sender As Object, ByVal e As DevExpress.XtraEditors.Controls.ButtonPressedEventArgs) Handles btnEdit_colSpacing.ButtonClick
			If myGridView1.FocusedColumn IsNot Nothing Then
				TryCast(myGridView1.FocusedColumn, MyGridColumn).ColumnSpacing = Convert.ToInt32(btnEdit_colSpacing.EditValue)
			End If
		End Sub

		Private Sub btnEdit_colSpacing_KeyDown(ByVal sender As Object, ByVal e As KeyEventArgs) Handles btnEdit_colSpacing.KeyDown
			If e.KeyData = Keys.Enter Then
				btnEdit_colSpacing_ButtonClick(Me, Nothing)
			End If
		End Sub

		Private Sub cbRowCellStyle_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbRowCellStyle.CheckedChanged
			myGridView1.LayoutChanged()
		End Sub
	End Class
End Namespace

File: CS/ColumnSpacing/Form1.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;


namespace ColumnSpacing
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            myGridControl1.DataSource = CreateTable(20);
            myGridView1.Columns[1].ColumnSpacing = 8;
            myGridView1.BestFitColumns();
        }

        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) });
            return tbl;
        }

        private void myGridView1_RowCellStyle(object sender, RowCellStyleEventArgs e)
        {
            if (cbRowCellStyle.Checked)
            {
                var view = myGridView1;
                if (e.RowHandle != view.FocusedRowHandle)
                {
                    if (e.RowHandle % 2 == 0)
                        e.Appearance.BackColor = Color.FromArgb(0xEB, 0xF9, 0xEB);
                    else
                        e.Appearance.BackColor = Color.FromArgb(0xED, 0xF9, 0xFA);
                }
                else
                    e.Appearance.BackColor = SystemColors.GradientInactiveCaption;
            }
        }

        private void myGridView1_FocusedColumnChanged(object sender, DevExpress.XtraGrid.Views.Base.FocusedColumnChangedEventArgs e)
        {
            btnEdit_colSpacing.EditValue = (e.FocusedColumn as MyGridColumn).ColumnSpacing;
        }

        private void btnEdit_colSpacing_ButtonClick(object sender, DevExpress.XtraEditors.Controls.ButtonPressedEventArgs e)
        {
            if (myGridView1.FocusedColumn != null)
                (myGridView1.FocusedColumn as MyGridColumn).ColumnSpacing = Convert.ToInt32(btnEdit_colSpacing.EditValue);
        }

        private void btnEdit_colSpacing_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyData == Keys.Enter) btnEdit_colSpacing_ButtonClick(this, null);
        }

        private void cbRowCellStyle_CheckedChanged(object sender, EventArgs e)
        {
            myGridView1.LayoutChanged();
        }
    }
}
